streams/shared/src/main/scala/zio/stream/ZSink.scala

```diff
@@ -658,7 +658,7 @@ trait ZSink[-R, +E, +A0, -A, +B] extends Serializable { self =>
                 if (self.cont(s2)) UIO.succeedNow(Left(s2))
                 else self.extract(s2).map(Right(_))
               },
-            { case (b, leftover) => UIO.succeedNow(Right((b, leftover ++ Chunk.single(a)))) }
```

Comment:
I don't think these change are needed.

Reply:
So far they are, I will check again an investigate.

Reply:
I am this error : 
```scala
[info] Compiling 1 Scala source to /Users/jon/Projects/zio/streams/jvm/target/scala-2.12/classes ...
[error] /Users/jon/Projects/zio/streams/shared/src/main/scala/zio/stream/ZSink.scala:661:86: type mismatch;
[error]  found   : zio.Chunk.NonEmpty[A1]
[error]  required: zio.Chunk.NonEmpty[A00]
[error]             { case (b, leftover) => UIO.succeedNow(Right((b, leftover ++ Chunk.single(a)))) }
[error]     
```

A00 and A1 are the same here, I am taking a deeper look.

Reply:
Yeah, I saw that error too when I was playing with it. Changing the `++` encoding resolved it.